packages/worker/src/workflow/reusable-vm.ts

```diff
@@ -104,39 +119,62 @@ export class ReusableVMWorkflowCreator implements WorkflowCreator {
    */
   async createWorkflow(options: WorkflowCreateOptions): Promise<Workflow> {
     const context = this.context;
-    const bag: Record<string, unknown> = {};
-    const { isolateExecutionTimeoutMs, contextKeysToPreserve } = this;
+    const pristine = this.pristine;
+    const bag: Map<string, PropertyDescriptor> = new Map();
+    const { isolateExecutionTimeoutMs } = this;
     const workflowModule: WorkflowModule = new Proxy(
       {},
       {
         get(_: any, fn: string) {
           return (...args: any[]) => {
-            Object.assign(context, bag);
+            for (const [pname, pdesc] of bag.entries()) {
+              Object.defineProperty(context, pname, pdesc);
+            }
```

Comment:
I'm a bit confused about`bag` initialization, in particular if the first time we call 
` return vm.runInContext(`__TEMPORAL__.api.${fn}``
the bag would be empty, and whether that matters.
I see the bag is filled inside  `finally{...` after `bag.clear()` but not sure about the first call...